pkg/cloud/scope/cluster.go

```diff
@@ -98,6 +101,28 @@ type ClusterScope struct {
 	controllerName  string
 }
 
+// KubeadmControlPlaneBindPort returns the bind port used in KCP kubeconfig.
+func (s *ClusterScope) KubeadmControlPlaneBindPort() (int32, error) {
+	if s.Cluster.Spec.ControlPlaneRef != nil {
+		cpObjectKey := client.ObjectKey{
+			Namespace: s.Cluster.Namespace,
+			Name:      s.Cluster.Spec.ControlPlaneRef.Name,
+		}
+		var kubeadmCP controlplanev1.KubeadmControlPlane
+		err := s.client.Get(context.TODO(), cpObjectKey, &kubeadmCP)
```

Comment:
Should we use `context.Background()` instead of `context.TODO()`. wdyt?

Reply:
IMO context.TODO() should be used, as we mostly use context.Background() from where the call starts, since this will come somewhere during reconciliation, we might already have the context in place, so we should use context.TODO()